f2/src/forms/WhenDidItHappenForm.js

```diff
@@ -76,68 +46,53 @@ export const WhenDidItHappenForm = (props) => {
     values.description = ''
   }
 
-  const formatData = (values) => {
-    let errors = {}
-
+  const clearData = (values) => {
     if (values.incidentFrequency === 'once') {
-      const happenedOnceError = evalDate(
-        values.happenedOnceDay,
-        values.happenedOnceMonth,
-        values.happenedOnceYear,
-      )
-
-      if (happenedOnceError) {
-        errors['happenedOnceError'] = happenedOnceError
-      } else {
-        clearDateRange(values)
-        clearDescription(values)
-      }
+      clearDateRange(values)
+      clearDescription(values)
     } else if (values.incidentFrequency === 'moreThanOnce') {
-      const dateRangeErrors = evalDateRange(values)
-
-      if (dateRangeErrors.startError) {
-        errors['startError'] = dateRangeErrors.startError
-      }
-
-      if (dateRangeErrors.endError) {
-        errors['endError'] = dateRangeErrors.endError
-      }
       clearHappenedOnce(values)
       clearDescription(values)
     } else {
       clearHappenedOnce(values)
       clearDateRange(values)
     }
-
-    return errors
   }
 
   return (
     <React.Fragment>
       <Formik
         initialValues={whenDidItHappen}
-        initialStatus={{ errors: '' }}
-        validationSchema={whenDidItHappenFormSchema()}
         validateOnChange={false}
-        onSubmit={(values, { setStatus }) => {
-          const errors = formatData(values)
-          setStatus({ errors })
-          if (Object.keys(errors).length === 0) {
+        validate={(values) => {
+          return realTimeValidation(values)
+        }}
+        onSubmit={async (values, { setErrors }) => {
+          const errors = onSubmitValidation(values)
+          if (errors.fields) {
+            setErrors(errors.fields)
+          } else {
+            await clearData(values)
             props.onSubmit(values)
           }
         }}
       >
-        {({ handleSubmit, handleChange, handleBlur, status, errors }) => (
+        {({ handleSubmit, handleChange, handleBlur, submitCount, errors }) => (
           <Form onSubmit={handleSubmit}>
-            <FormikListener />
             <Container>
               <Row className="form-question">
-                <Row className="form-label">
+                {Object.keys(errors).length > 0 && (
```

Comment:
Is it a good idea to check errors.size inside ErrorSummary component? Instead of wrapping the check outside?

Reply:
Due to use of hooks in the component this would require further refactoring. Not a bad idea though.